src/main/java/org/jboss/xavier/integrations/route/VMWorkloadInventoryRoutes.java

```diff
@@ -41,42 +58,18 @@ public void configure() throws Exception {
             .transform().method("decisionServerHelper", "extractWorkloadInventoryReportModel");
 
         from("direct:flags-shared-disks").routeId("flags-shared-disks")
+                .process(exchange -> {
+                    String originalBody = exchange.getIn().getBody(String.class);
+                    exchange.getIn().setHeader("originalBody", originalBody);
+                })
             .bean("flagSharedDisksCalculator", "calculate(${body}, ${header.${type:org.jboss.xavier.integrations.route.MainRouteBuilder.MA_METADATA}})", false)
             .process(exchange -> {
+                String originalBody = exchange.getIn().getHeader("originalBody", String.class);
                 Set<String> vmNamesWithSharedDisk = exchange.getIn().getBody(Set.class);
-                List<WorkloadInventoryReportModel> workloadInventoryReportModels = workloadInventoryReportService.findByAnalysisOwnerAndAnalysisId(
-                        exchange.getIn().getHeader(USERNAME, String.class),
-                        Long.parseLong(exchange.getIn().getHeader(MA_METADATA, Map.class).get(ANALYSIS_ID).toString()));
-                List<WorkloadInventoryReportModel> workloadInventoryReportModelsToUpdate = workloadInventoryReportModels.stream()
-                    .filter(workloadInventoryReportModel -> vmNamesWithSharedDisk.contains(workloadInventoryReportModel.getVmName()))
-                    .peek(workloadInventoryReportModel -> workloadInventoryReportModel.addFlagIMS("Shared Disk")).collect(Collectors.toList());
-                exchange.getIn().setBody(workloadInventoryReportModelsToUpdate);
-            })
-            .to("direct:reevaluate-workload-inventory-reports");
+                exchange.getIn().setHeader("vmNamesWithSharedDisk", vmNamesWithSharedDisk);
+                exchange.getIn().setBody(originalBody);
+            }).to("direct:calculate-vmworkloadinventory");
```

Comment:
Maybe I'm missing something but the code inside `direct:flags-shared-disks` is not persisted or used anywhere, not even in the next route `direct:calculate-vmworkloadinventory`. I've just created a PR to your PR https://github.com/m-brophy/xavier-integration/pull/2 to be more clear what I mean. I've tested my PR in CI and I have the same results as with your PR. Could you please take a look at my PR and verify if removing this block of code does not affect the logic?

Reply:
The names of vms which use a shared disk are stored in the header which is then retrieved in the next route and used to determine which vms have .setHasSharedVmdk(true) applied to them. 

To test this block of code you need to use a payload which would have triggered the old SHARED_DISK flag via the removed reevaulate route and ensure you still get shared_vmdk flags for the appropriate vms even though there is no use_shared_vmdk: true element in the payload